### PR TITLE
support get logic ddl sql for mysql

### PR DESCRIPTION
**Pull request by tuichenchuxin**
For #17137.

Changes proposed in this pull request:
- support get logic ddl sql for mysql

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import javax.sql.DataSource;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * DDL SQL generator for MySql.
+ */
+public final class MySqlDDLSQLGenerator implements DialectDDLSQLGenerator {
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java:31]: @tuichenchuxin Please rename it to MySQLDDLSQLGenerator.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    
+    private static final String SHOW_CREATE_SQL = "SHOW CREATE TABLE %s";
+    
+    @Override
+    public String generateDDLSQL(final String tableName, final String schemaName, final DataSource dataSource) throws SQLException {
+        try (
+                Statement statement = dataSource.getConnection().createStatement();
+                ResultSet resultSet = statement.executeQuery(String.format(SHOW_CREATE_SQL, tableName))) {
+            if (resultSet.next()) {
+                return resultSet.getString(2);
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java:41]: Can we use resultSet.getString(columnName)？

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            }
+        }
+        throw new ShardingSphereException(String.format("Failed to get ddl sql for table %s", tableName));
```
Comment (strongduanmu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java:44]: Please remove String.format here.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java (hunk 2 above)
Comment (tuichenchuxin) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java]: Thank you. Done.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java (hunk 3 above)
Comment (tuichenchuxin) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java]: Thank you. Done.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java (hunk 1 above)
Comment (tuichenchuxin) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ddlgenerator/MySqlDDLSQLGenerator.java]: Thank you. Done.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**